### ISSUE TITLE
`conda_run_command` doesn't use the `hatch-conda`  generated environment

### ISSUE THREAD

**Issue by jmuchovej** [OldGrumpyViking/hatch-conda#2]
I was trying to use `hatch-conda` earlier and found that the generated command (when doing `hatch run <script>`) wasn't specifying the correct `conda` environment, so it ends up running in `base`, which (of course) may not have the correct packages/dependencies installed.

https://github.com/OldGrumpyViking/hatch-conda/blob/99e0dded8785ffbacc358f2ef3ef1205411a6c63/hatch_conda/plugin.py#L103-L104

At face-value, reading these lines tells me everything should be generated correctly, yet when I do `hatch run <script>`, I end up with:
```shell
ERROR conda.cli.main_run:execute(49): `conda run <script>` failed. (See above for error)
```

Activating the `hatch-conda`-generated env also shows that the deps are missing, so I'm not sure how to go about fixing this. 😅

**Comment by OldGrumpyViking** [comment]
I am unable to recreate what you describe. Using the provided pyproject.toml file I get a mamba environment called `<proj_name>_default_3.9` and looking at the installed packages in the environment everything seems correct.

I tested it on a linux which I assume you are using based on your other issue :)

However I am unable to run your script command with the given dependencies as it cannot find the builders module.

```console
$ python -c 'from builders.python import mkgrammar; mkgrammar()'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'builders'
```

As hatch is quite new it still has some problems. I have experiences that if an environment fails in the creation process once it can be broken forever. Maybe you can try running `hatch env prune` and then try again with `hatch run pip list` to check if it worked.

**Comment by jmuchovej** [comment]
OH. I didn't realize you were planning to run this based on the `pyproject.toml`. `builders` is a local module. Running `python -c 'from builders.python import mkgrammar; mkgrammar()'` in an environment with `antlr4` installed works – I've tried in various `conda` envs and even `virtualenv`s.

I'll look into pruning `hatch` envs later today once I'm done writing up this tool. 😅 

Will report back once I've tested this. 🙂
